python/components/deframers/geoscan_deframer.py

```diff
@@ -73,3 +72,7 @@ def add_options(cls, parser):
             '--syncword_threshold', type=int,
             default=cls._default_sync_threshold,
             help='Syncword bit errors [default=%(default)r]')
+        parser.add_argument(
+            '--packet_length', type=int,
+            default=cls._default_packet_length,
+            help='Frame length in bytes [default=%(default)r]')
```

Comment:
Packet length shouldn't be handled via command line options. The SatYAML file should contain `framing`s for all the packet lengths that the satellite can transmit. The user should not need to specify a packet length with command line options.

Reply:
The thing is that we don’t yet know exactly what the lengths might be, they can be completely different and change over time.

Reply:
If the packets do not include a length field this means that the receiver must know the packet length ahead of time. Therefore, the SatYAML file will need to be updated any time that the satellite transmitter configuration changes. This is unfortunate, because of the need to keep updating this file, but the responsibility of using the correct packet length cannot lie within the users of gr-satellites. Command line parameters are to adjust receiver parameters that are not directly related to the transmitter modulation and coding configuration, and that might need to be adjusted for different SNRs, clock stabilities, etc. (examples: loop bandwidths and syncword thresholds).

There is already a `frame size` parameter in SatYAML ([example](https://github.com/daniestevez/gr-satellites/blob/main/python/satyaml/EIRSAT-1.yml#L12)). This is used by the CCSDS deframers and also by other deframers, such as `SMOG-P RA`. The same `frame size` parameter should also be used here with the GEOSCAN deframer.

Reply:
To be honest, I don't quite understand how to add the frame size field to the yml file code. How can I do this?

Reply:
I think you might want to take a look at this

Reply:
At what? The changes in 17cddff look good, but there are still some comments pending (mostly about documentation) and pycodestyle doesn't pass.